tests/js/server/dump/dump-mixed.js

```diff
@@ -0,0 +1,553 @@
+      assertEqual(0, c.count());
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test autoinc keygen
+////////////////////////////////////////////////////////////////////////////////
+//
+//    testKeygenAutoInc : function () {
+//      var c = db._collection("UnitTestsDumpKeygen");
+//      var p = c.properties();
+//
+//      assertEqual(2, c.type()); // document
+//      assertFalse(p.waitForSync);
+//      assertEqual("autoincrement", p.keyOptions.type);
+//      assertEqual(42049, p.keyOptions.lastValue);
+//      assertFalse(p.keyOptions.allowUserKeys);
+//      assertEqual(7, p.keyOptions.offset);
+//      assertEqual(42, p.keyOptions.increment);
+//
+//      assertEqual(1, c.getIndexes().length); // just primary index
+//      assertEqual("primary", c.getIndexes()[0].type);
+//      assertEqual(1001, c.count());
+//
+//      for (var i = 0; i < 1000; ++i) {
+//        var doc = c.document(String(7 + (i * 42)));
+//
+//        assertEqual(String(7 + (i * 42)), doc._key);
+//        assertEqual(i, doc.value);
+//        assertEqual({ value: [ i, i ] }, doc.more);
+//      }
+//      doc = c.save({});
+//      assertEqual(doc._key, "42091");
+//    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test padded keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenPadded : function () {
+      var c = db._collection("UnitTestsDumpKeygenPadded");
+      var p = c.properties();
+
+      assertEqual(2, c.type()); // document
+      assertFalse(p.waitForSync);
+      assertEqual("padded", p.keyOptions.type);
+      assertEqual(process.env['paddedLastValue'], p.keyOptions.lastValue);
+      assertFalse(p.keyOptions.allowUserKeys);
+      assertEqual(1, c.getIndexes().length); // just primary index
+      assertEqual("primary", c.getIndexes()[0].type);
+      assertEqual(1001, c.count());
+
+      let allDocs = {};
+      let nonDeletedDoc;
+      c.toArray().forEach(doc => {
+        if (doc.hasOwnProperty('value')) {
+          allDocs[doc.value] = doc;
+        } else {
+          nonDeletedDoc = doc;
+        }
+      });
+      
+      let lastKey = "";
+      for (var i = 0; i < 1000; ++i) {
+        var doc = allDocs[i];
+
+        assertTrue(doc._key > lastKey, doc._key + ">" + lastKey);
+        assertEqual(i, doc.value);
+        assertEqual({ value: [ i, i ] }, doc.more);
+        lastKey = doc._key;
+      }
+      assertTrue(nonDeletedDoc._key > lastKey, nonDeletedDoc._key + ">" + lastKey);
+      lastKey = nonDeletedDoc._key;
+      nonDeletedDoc = c.save({});
+      assertTrue(nonDeletedDoc._key > lastKey, nonDeletedDoc._key + ">" + lastKey);
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test uuid keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenUuid : function () {
```

Comment:
The same (or almost the same) method I've just reviewed in "tests/js/server/dump/dump-mixed-cluster.js". 
Please re-use that method here. This is hard to maintain.